src/AbstractPhpdocToTypeDeclarationFixer.php

```diff
@@ -208,6 +204,75 @@ protected function getCommonTypeFromAnnotation(Annotation $annotation, bool $isR
         return [$commonType, $isNullable];
     }
 
+    protected function getUnionTypes(TypeExpression $typesExpression, bool $isReturnType): ?string
+    {
+        if (\PHP_VERSION_ID < $this->versionSpecificTypes['union-types']) {
+            return null;
+        }
+
+        if (!$typesExpression->isUnionType() || '|' !== $typesExpression->getTypesGlue()) {
```

Comment:
intersect types are also supported by the latest PHP, so this PR should aim to support it as well

Reply:
Yes, but I believe it is only nice to have here, since the PR explicitly states that it provides support for union types. It could be added later, probably along with refactoring related to naming in `TypeExpression` ([see here](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/6359#discussion_r1377385303)).

Reply:
Yes and no, well code PR should not be much bigger when supporting both...

Reply:
Extracted to separate issue, will try to work on this.